### ISSUE TITLE
.~lock.<filename>.docx#

### ISSUE THREAD

**Issue by kugelpunk** [typhoon-hil/LibreOfficeToPDF#9]
Hi,

I can't seem to avoid the above temp file being created along with the PDF. Would you be able to clarify (for Windows-users)?

Thanks in advance
Great project!

**Comment by kugelpunk** [comment]
Is it a known bug, I mean is it standard behavior, as it is typically no problem? I am using LibreOfficPortable (6.2.5.2) on a server creating loads of files.

**Comment by Sup3rGeo** [comment]
Hi! This file is probably created by LibreOffice itself as the library does not intentionally creates any extra files.

**Comment by kugelpunk** [comment]
Yep, it's the standard lock used for making LibreOffice-files inaccessible for other processes while opened. In my implementation LibreOffice does not properly close the docx-file after printing. I am trying to see if someone else has the same problem, if we could keep the issue open.

**Comment by kugelpunk** [comment]
This solution works fine for me:
https://forum-test.openoffice.org/en/forum/viewtopic.php?f=25&t=75367#p343573
```python
    # Avoid open file error and lock-files (.~lock.filename.docx#)
    if document.supportsService("com.sun.star.util.XCloseable"):
        document.close(True)
    document.dispose()
```
It needs to be included here:
https://github.com/typhoon-hil/LibreOfficeToPDF/blob/ded1a7514df0b37662048d710c6ae938495fa3cf/src/LibreOfficeToPDF/script.py#L185

**Comment by kugelpunk** [comment]
Kept open for sake of this repo.
